.github/workflows/check-release.yml

```diff
@@ -0,0 +1,170 @@
+      ## NOTE: The SHA obtained in this step differs from the previous one, as it represents the branch commit SHA used for change verification.
+
+      - name: Get Docker image digests
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        id: get_digests
+        run: |
+          digests=$(gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /orgs/actions/packages/container/actions-runner/versions)
+          latest_digest=$(echo "$digests" | jq -r '.[] | select(.metadata.container.tags | index("latest")) | .name')
+          echo $latest_digest
+          echo "arch_sha_name=$latest_digest" >> "$GITHUB_OUTPUT"
+        env:
+          GITHUB_TOKEN: ${{ env.GITHUB_TOKEN }}
+
+      - name: Get changelog file
+        id: changelog
+        run: |
+          if [ ! -f CHANGELOG.md ]; then
+            echo "Changelog file does not exist. Creating one..."
+            touch CHANGELOG.md
+          fi
+
+      - name: Update Changelog
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        id: update_changelog
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const changelogPath = 'CHANGELOG.md';
+            const dockerImage = `docker pull ghcr.io/actions/runner:${{ steps.release_outputs.outputs.tag_name }}@${{ steps.get_digests.outputs.arch_sha_name }}`;
+  
+            const changelogEntry = `## ${{ steps.release_outputs.outputs.tag_name }}\n- Docker Image: \`${dockerImage}\`\n`;
+  
+            const changelogContent = fs.readFileSync(changelogPath, 'utf8');
+            fs.writeFileSync(changelogPath, `${changelogEntry}\n${changelogContent}`);
+    
+      - name: Update Dockerfile
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        id: update_dockerfile
+        run: |
+          echo "Updating Dockerfile..."
+          
+          # Construct the new FROM statement
+          new_from="FROM ghcr.io/actions/actions-runner:${{ steps.release_outputs.outputs.short_tag_name }}@${{ steps.get_digests.outputs.arch_sha_name }} AS base"
+          
+          # Replace the first line with the new FROM statement
+          sed -i "1s|.*|$new_from|" Dockerfile
+
+          echo "Dockerfile updated."
+      
+      ## NOTE: To permit the PR creation the option "Allow GitHub Actions to create and approve pull requests" must be enabled in Repository Settings -> Actions -> General.
+      # Reference: https://github.com/peter-evans/create-pull-request
+      #            https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request?tool=cli
+      - name: Create PR
+        if: steps.check_release_changes.outputs.release_changed == 'true'
+        uses: peter-evans/create-pull-request@v6
```

Comment:
Not sure we can use this action

Reply:
How can we verify it? In any case, we can convert it into a script step that executes git commands.

Reply:
I agree that it would be better to use `gh pr`